shelley/chain-and-ledger/executable-spec/src/Keys.hs

```diff
@@ -240,3 +250,35 @@ hashKeyES
   -> KeyHashES hashAlgo kesAlgo
 hashKeyES (VKeyES vKeyES) =
   KeyHashES $ hashWithSerialiser encodeVerKeyKES vKeyES
+
+-- | Our VRFAlgorithm provides a 'Natural', so we must exhibit an isomorphism to
+--   use the bits as some other type
+class VRFValue a where
+  fromNatural :: Natural -> a
+  toNatural :: a -> Natural
+
```

Comment:
using `Hash` as suggested above should also be able to do a conversion from / to Natural